wrouter/router_root.go

```diff
@@ -202,6 +209,28 @@ func (r *rootRouter) AddRouteHandlerMiddleware(handlers ...RouteHandlerMiddlewar
 	r.routeHandlers = append(r.routeHandlers, handlers...)
 }
 
+func (r *rootRouter) RegisterNotFoundHandler(handler http.Handler) {
+	wrappedHandlerFn := createRouteRequestHandler(func(rw http.ResponseWriter, r *http.Request, reqVals RequestVals) {
+		handler.ServeHTTP(rw, r)
+	}, r.routeHandlers)
+
+	r.impl.RegisterNotFoundHandler(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		wrappedHandlerFn(rw, r, RequestVals{
+			Spec: RouteSpec{
+				Method:       r.Method,
+				PathTemplate: "/*",
```

Comment:
I tested this on a java service, and they are able to inject the subrouter's context path if one matches. E.g. `/foo/api/bad/path` would generate `/foo/api/*`. This is difficult for us because the subrouters are flattened when provided to the router impl, so I think this is fine.